### PR TITLE
Simplify and improve DropDownGlyph in Fluent ComboBox

### LINKED ISSUE #7452: Fluent ComboBox DropDownGlyph Improvements.
**Is your feature request related to a problem? Please describe.**

No, just refactoring.

**Describe the solution you'd like**

The Fluent ComboBox needs to be revisited after work on the SplitButton. Specifically, the DropDownGlyph needs to be looked at:

```xaml
            <Viewbox UseLayoutRounding="False"
                     MinHeight="{DynamicResource ComboBoxMinHeight}"
                     Grid.Row="1"
                     Grid.Column="1"
                     IsHitTestVisible="False"
                     Margin="0,0,10,0"
                     Height="12"
                     Width="12"
                     HorizontalAlignment="Right"
                     VerticalAlignment="Center">
              <Panel>
                <Panel Height="12"
                       Width="12" />
                <Path x:Name="DropDownGlyph"
                      VerticalAlignment="Center" />
              </Panel>
            </Viewbox>
```

Additionally, all these drop-down glyphs can be pulled out into a separate resource and shared among all controls that need them. That will allow swapping the glyph out if needed. Full discussion is buried in the SplitButton PR.

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by robloo**
## What does the pull request do?

Simplifies and improves the DropDownGlyph in the ComboBox's fluent Style.

Closes #7452 

## What is the current behavior?

The code looks really hacked together after the initial port from WinUI of the style. At the time I don't think PathIcon existed in Avalonia either.

```xaml
            <Viewbox UseLayoutRounding="False"
                     MinHeight="{DynamicResource ComboBoxMinHeight}"
                     Grid.Row="1"
                     Grid.Column="1"
                     IsHitTestVisible="False"
                     Margin="0,0,10,0"
                     Height="12"
                     Width="12"
                     HorizontalAlignment="Right"
                     VerticalAlignment="Center">
              <Panel>
                <Panel Height="12"
                       Width="12" />
                <Path x:Name="DropDownGlyph"
                      VerticalAlignment="Center" />
              </Panel>
            </Viewbox>

```

Visually it looked fine:

image

## What is the updated/expected behavior with this PR?

Code is cleaned up and made more sensible.

```xaml
            <PathIcon x:Name="DropDownGlyph"
                      Grid.Row="1"
                      Grid.Column="1"
                      UseLayoutRounding="False"
                      IsHitTestVisible="False"
                      Height="12"
                      Width="12"
                      Margin="0,0,10,0"
                      HorizontalAlignment="Right"
                      VerticalAlignment="Center" />
```

Visually it looks the same:

image

It now is much closer to Fluent v1 style as well:

https://github.com/microsoft/microsoft-ui-xaml/blob/09267531b807707138addc9bef3c8e9fa340615a/dev/ComboBox/ComboBox_themeresources_v1.xaml#L695-L707

## How was the solution implemented (if it's not obvious)?

Described above.

## Checklist

- [ ] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Documentation with user documentation

## Breaking changes

 - Anyone re-templating will have to target `Foreground` instead of `Fill` for the DropDownGlyph. The type is also changing from `Path` to `PathIcon`

## Fixed issues

Fixes #7452

### CODE REVIEW

**Review by jmacato — APPROVED** [review]
LGTM

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0019377-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

File: src/Avalonia.Themes.Fluent/Controls/ComboBox.xaml [hunk 1]
```diff
@@ -159,10 +153,9 @@
     <Setter Property="IsVisible" Value="False" />
   </Style>
 
-  <Style Selector="ComboBox /template/ Path#DropDownGlyph">
-    <Setter Property="Fill" Value="{DynamicResource ComboBoxDropDownGlyphForeground}" />
+  <Style Selector="ComboBox /template/ PathIcon#DropDownGlyph">
+    <Setter Property="Foreground" Value="{DynamicResource ComboBoxDropDownGlyphForeground}" />
```
Comment (robloo) [inline src/Avalonia.Themes.Fluent/Controls/ComboBox.xaml:157]: Strangely this does not work. I can change Data property here just fine but the `Foreground` property is ignored completely. I can set it to `Red` here and there will be no effect.

If the `Foreground` is set above in the `PathIcon` declaration it works fine. If `Foreground` is set within the template of `PathIcon` itself it works fine. I can't imagine what priority-type issue is causing the issue here...

**Comment by robloo** [comment]
@jmacato Well, thanks for looking at it so quickly. I found a new issue though. Please see my review comments. I'll have to open a new PR to fix it then...

File: src/Avalonia.Themes.Fluent/Controls/ComboBox.xaml (hunk 1 above)
Comment (robloo) [reply src/Avalonia.Themes.Fluent/Controls/ComboBox.xaml:157]: Addressed #7859

**Pull request merged**